core/test/base/batch_lin_op.cpp

```diff
@@ -0,0 +1,164 @@
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+******************************<GINKGO LICENSE>*******************************/
+
+#include <ginkgo/core/base/batch_lin_op.hpp>
+
+
+#include <complex>
+#include <memory>
+#include <type_traits>
+
+
+#include <gtest/gtest.h>
+
+
+#include <ginkgo/core/base/executor.hpp>
+#include <ginkgo/core/base/lin_op.hpp>
+#include <ginkgo/core/base/math.hpp>
+
+
+namespace {
+
+
+class DummyBatchLinOp : public gko::batch::EnableBatchLinOp<DummyBatchLinOp>,
+                        public gko::EnableCreateMethod<DummyBatchLinOp> {
+public:
+    DummyBatchLinOp(std::shared_ptr<const gko::Executor> exec,
+                    gko::batch_dim<2> size = gko::batch_dim<2>{})
+        : gko::batch::EnableBatchLinOp<DummyBatchLinOp>(exec, size)
+    {}
+};
+
+
+class EnableBatchLinOp : public ::testing::Test {
+protected:
+    EnableBatchLinOp()
+        : ref{gko::ReferenceExecutor::create()},
+          ref2{gko::ReferenceExecutor::create()},
+          op{DummyBatchLinOp::create(ref2,
+                                     gko::batch_dim<2>(1, gko::dim<2>{3, 5}))},
+          op2{DummyBatchLinOp::create(ref2,
+                                      gko::batch_dim<2>(2, gko::dim<2>{3, 5}))}
+    {}
+
+    std::shared_ptr<const gko::ReferenceExecutor> ref;
+    std::shared_ptr<const gko::ReferenceExecutor> ref2;
+    std::unique_ptr<DummyBatchLinOp> op;
+    std::unique_ptr<DummyBatchLinOp> op2;
+};
+
+
+TEST_F(EnableBatchLinOp, KnowsNumBatchItems)
+{
+    ASSERT_EQ(op->get_num_batch_items(), 1);
+    ASSERT_EQ(op2->get_num_batch_items(), 2);
+}
+
+
+TEST_F(EnableBatchLinOp, KnowsItsSizes)
+{
+    auto op1_sizes = gko::batch_dim<2>(1, gko::dim<2>{3, 5});
+    auto op2_sizes = gko::batch_dim<2>(2, gko::dim<2>{3, 5});
+    ASSERT_EQ(op->get_size(), op1_sizes);
+    ASSERT_EQ(op2->get_size(), op2_sizes);
+}
+
+
+template <typename T = int>
+class DummyBatchLinOpWithFactory
+    : public gko::batch::EnableBatchLinOp<DummyBatchLinOpWithFactory<T>> {
+public:
+    DummyBatchLinOpWithFactory(std::shared_ptr<const gko::Executor> exec)
+        : gko::batch::EnableBatchLinOp<DummyBatchLinOpWithFactory>(exec)
+    {}
+
+    GKO_CREATE_FACTORY_PARAMETERS(parameters, Factory)
+    {
+        T GKO_FACTORY_PARAMETER_SCALAR(value, T{5});
+    };
+    GKO_ENABLE_BATCH_LIN_OP_FACTORY(DummyBatchLinOpWithFactory, parameters,
+                                    Factory);
+    GKO_ENABLE_BUILD_METHOD(Factory);
+
+    DummyBatchLinOpWithFactory(const Factory* factory,
+                               std::shared_ptr<const gko::batch::BatchLinOp> op)
+        : gko::batch::EnableBatchLinOp<DummyBatchLinOpWithFactory>(
+              factory->get_executor()),
+          parameters_{factory->get_parameters()},
+          op_{op}
+    {}
+
+    std::shared_ptr<const gko::batch::BatchLinOp> op_;
```

Comment:
```suggestion
    std::shared_ptr<const gko::batch::BatchLinOp> op;
```
nit: because it is a public member

Reply:
I think it is okay in the tests here as it makes it easier to differentiate that it is a member of the `op` object.